src/parser/lexer.rs

```diff
@@ -189,69 +187,65 @@ fn lex_comment(input: &str) -> LexResult {
-            map(tag("EXCHANGE"), |_| Exchange),
-            map(tag("GE"), |_| GE),
-            map(tag("GT"), |_| GT),
-            map(tag("IOR"), |_| Ior),
-            map(tag("LE"), |_| LE),
-            map(tag("LOAD"), |_| Load),
-            map(tag("LT"), |_| LT),
-            map(tag("MOVE"), |_| Move),
-            map(tag("MUL"), |_| Mul),
-            map(tag("NEG"), |_| Neg),
-            map(tag("NOT"), |_| Not),
-            map(tag("STORE"), |_| Store),
-            map(tag("SUB"), |_| Sub),
-            map(tag("XOR"), |_| Xor),
-        )),
-        alt((
-            value(Capture, tag("CAPTURE")),
-            value(DefCal, tag("DEFCAL")),
-            value(DefFrame, tag("DEFFRAME")),
-            value(DefWaveform, tag("DEFWAVEFORM")),
-            value(Delay, tag("DELAY")),
-            value(Fence, tag("FENCE")),
-            value(Pulse, tag("PULSE")),
-            value(RawCapture, tag("RAW-CAPTURE")),
-            value(SetFrequency, tag("SET-FREQUENCY")),
-            value(SetPhase, tag("SET-PHASE")),
-            value(SetScale, tag("SET-SCALE")),
-            value(ShiftFrequency, tag("SHIFT-FREQUENCY")),
-            value(ShiftPhase, tag("SHIFT-PHASE")),
-        )),
-    ))(input)?;
-    Ok((result.0, Token::Command(result.1)))
+    let command = match identifier.as_str() {
+        "DEFGATE" => Some(DefGate),
+        "ADD" => Some(Add),
+        "AND" => Some(And),
+        "CONVERT" => Some(Convert),
+        "DIV" => Some(Div),
+        "EQ" => Some(Eq),
+        "EXCHANGE" => Some(Exchange),
+        "GE" => Some(GE),
+        "GT" => Some(GT),
+        "IOR" => Some(Ior),
+        "LE" => Some(LE),
+        "LOAD" => Some(Load),
+        "LT" => Some(LT),
+        "MOVE" => Some(Move),
+        "MUL" => Some(Mul),
+        "NEG" => Some(Neg),
+        "NOT" => Some(Not),
+        "STORE" => Some(Store),
+        "SUB" => Some(Sub),
+        "XOR" => Some(Xor),
+        "DEFCIRCUIT" => Some(DefCircuit),
+        "MEASURE" => Some(Measure),
+        "HALT" => Some(Halt),
+        "JUMP-WHEN" => Some(JumpWhen),
+        "JUMP-UNLESS" => Some(JumpUnless),
+        "JUMP" => Some(Jump),
+        "RESET" => Some(Reset),
+        "WAIT" => Some(Wait),
+        "NOP" => Some(Nop),
+        "INCLUDE" => Some(Include),
+        "PRAGMA" => Some(Pragma),
+        "DECLARE" => Some(Declare),
+        "CAPTURE" => Some(Capture),
+        "DEFCAL" => Some(DefCal),
+        "DEFFRAME" => Some(DefFrame),
+        "DEFWAVEFORM" => Some(DefWaveform),
+        "DELAY" => Some(Delay),
+        "FENCE" => Some(Fence),
+        "PULSE" => Some(Pulse),
+        "RAW-CAPTURE" => Some(RawCapture),
+        "SET-FREQUENCY" => Some(SetFrequency),
+        "SET-PHASE" => Some(SetPhase),
+        "SET-SCALE" => Some(SetScale),
+        "SHIFT-FREQUENCY" => Some(ShiftFrequency),
+        "SHIFT-PHASE" => Some(ShiftPhase),
+        "LABEL" => Some(Label),
+        _ => None,
+    };
+
+    match command {
```

Comment:
Probably fine, but shadowing the word `command` here took me a second to mentally parse

Reply:
True - agreed that this was confusing, applied a simple fix to use a single `match`.